images/go-ipfs/tests/docker-test.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+CONTAINER_NAME="ipfs-host-$(uuidgen)"
+CONTAINER_PORT=${FREE_PORT}
+REQUEST_RETRIES=5
+RETRY_DELAY=15
+IPFS_PATH=/data/ipfs
+user=ipfs
+repo="$IPFS_PATH"
+
+
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# Log entries which we expect to be present when the service is running healthy.
+expected_logs=(
+  "Daemon is ready"
+)
+
+missing_logs=()
+ 
+# Start up a new IPFS container
+TEST_start_container() {
+  container_id=$(docker run \
+    -d \
+    --name "${CONTAINER_NAME}" -e IPFS_PROFILE=server -p 4001:4001 -p 4001:4001/udp -p 127.0.0.1:8080:8080 -p "${CONTAINER_PORT}:5001" \
+    "${IMAGE_NAME}" log tail)
+
+  trap "docker stop ${container_id}" EXIT
+  sleep 15
+  
+  if ! docker ps --filter "name=${CONTAINER_NAME}" --format '{{.Names}}' | grep -q "^${CONTAINER_NAME}$"; then
+    echo "FAILED: ${CONTAINER_NAME} is not running."
+    exit 1
+  fi
+}
+
+# To initialize configuration files and generate a new keypair
+SET_ENV(){
```

Comment:
This script isn't running in a container, so it isn't testing the built image. It likely isn't necessary anyways, a couple docker run commands with a shared volume will accomplish the same thing.